src/plugin.js

```diff
@@ -94,13 +94,19 @@ class InstallPlugin {
     const dep = installer.check(result.request);
 
     if (dep) {
-      let { dev } = this.options;
+      const { packageManagerOptions } = this.options;
 
-      if (typeof this.options.dev === 'function') {
-        dev = !!this.options.dev(result.request, result.path);
+      if (typeof packageManagerOptions.dev === 'function') {
```

Comment:
in this case, `packageManagerOptions` can be function (i.e. `return { dev: true | false }`), so developer can decide it will be as dev deps or as basic

Reply:
Like this -`packageManagerOptions: { dev? } | () => {}`

Reply:
`packageManagerOptions: { dev? } | (request, path): { dev: boolean } => {}`